### PR TITLE
Add Seq2seq example

### PR DESCRIPTION
**Pull request by keisukefukuda**

### CODE REVIEW

File: examples/seq2seq/seq2seq.py [hunk 1]
```diff
@@ -0,0 +1,544 @@
...
+        print('==========================================')
+        print('Num process (COMM_WORLD): {}'.format(MPI.COMM_WORLD.Get_size()))
+        if args.gpu:
+            print('Using GPUs')
+        print('Using {} communicator'.format(args.communicator))
+        print('Num unit: {}'.format(args.unit))
+        print('Num Minibatch-size: {}'.format(args.batchsize))
+        print('==========================================')
+
+    if False:
```
Comment (iwiwi) [inline examples/seq2seq/seq2seq.py]: Just in case: remove this before merge

File: examples/seq2seq/seq2seq.py [hunk 2]
```diff
@@ -0,0 +1,544 @@
...
+                                         # 'validation/main/loss',
+                                         'main/perp',
+                                         'validation/main/bleu',
+                                         # 'validation/main/perp',
+                                         'elapsed_time'])
+        trainer.extend(report, trigger=(1, 'epoch'))
+        # trainer.extend(translate, trigger=(200, 'iteration'))
+
+        # trainer.extend(CalculateBleu(model, test_data),
+        #                trigger=(10, 'iteration'))
```
Comment (iwiwi) [inline examples/seq2seq/seq2seq.py]: Please remove unused code

File: examples/seq2seq/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+Next, you need to install required packages.
+
+```
+$ pip install nltk progressbar2
+
+## Run
+
+```bash
+$ cd $CHAINERMN
```
[the remark this answers is not in the document]
Comment (keisukefukuda) [reply examples/seq2seq/README.md]: fixed!

**Comment by iwiwi** [comment]
Do we need `seq2seq.orig.py` here?

**Comment by iwiwi** [comment]
LGTM

**Pull request merged**